### PR TITLE
feat: hooks: `disabled_modules`

### PR DESCRIPTION
**Pull request by ssiyad**
This PR introduces a hook `blocked_modules`, which can be used to disable modules globally.

### CODE REVIEW

**Comment by phot0n** [comment]
will this be raised in upstream as well? since this is a feature, say a partner wants to block some additional stuff how would they come to know of this?

anyway i'm just rambling on :P

**Comment by ssiyad** [comment]
> will this be raised in upstream as well?

Is this useful/generic enough to be on upstream?

**Comment by phot0n** [comment]
I think so - I'm pretty sure its a common enough use case - in the case it isn't, reviewers will probably let you know :)

File: frappe/utils/user.py [hunk 1]
```diff
@@ -88,8 +89,13 @@ def build_doctype_map(self):
 		)
 
 		for dt in all_doctypes:
-			if not dt.restrict_to_domain or (dt.restrict_to_domain in active_domains):
-				self.doctype_map[dt["name"]] = dt
+			if dt.restrict_to_domain:
+				continue
+			if dt.restrict_to_domain in active_domains:
+				continue
```
[the remark this answers is not in the document]
Comment (ssiyad) [reply frappe/utils/user.py:95]: It indeed is wrong. I was just experimenting with it. Will fix.

File: frappe/desk/desktop.py [hunk 2]
```diff
@@ -428,9 +428,6 @@ def get_workspace_sidebar_items():
 		"module": ["not in", blocked_modules],
 	}
 
-	if has_access:
-		filters = []
-
```
Comment (phot0n) [inline frappe/desk/desktop.py:433]: this is a bit problematic btw - there are some workspaces which are "restricted to domains" which are not present in allow modules section in the user doctype and tend to not show up in the home page (/app)

eg: Retail

Comment (ssiyad) [reply frappe/desk/desktop.py:433]: I thought 'Restrict to Domains` was almost deprecated? Do we use that feature?

Comment (ssiyad) [reply frappe/desk/desktop.py:433]: https://github.com/ivendnext/iVendFramework/pull/9

**Pull request merged**